File: php/src/Commands/Status.php
<?php

declare(strict_types=1);

namespace FormHandle\Commands;

use FormHandle\{Api, Config, Output};

class Status
{
    public static function run(array $ctx): void
    {
        $res = Api::get('/');
        $config = Config::read();

        if ($ctx['json']) {
            Output::json(['api' => $res['data'], 'config' => $config]);
            return;
        }

        Output::heading('FormHandle API');

        if ($res['status'] === 200) {
            Output::success('API is reachable');
            if (isset($res['data']['status'])) {
                Output::info("Status: {$res['data']['status']}");
            }
            if (isset($res['data']['version'])) {
                Output::info("Version: {$res['data']['version']}");
            }
        } else {
            Output::error('API returned an unexpected status');
        }

        if ($config) {
            Output::heading('Local Config (.formhandle)');
            $domains = array_keys($config);
            foreach ($domains as $i => $domain) {
                $ep = $config[$domain];
                echo "  $domain\n";
                Output::table([
                    ['handler_id', $ep['handler_id'] ?? ''],
                    ['email', $ep['email'] ?? ''],
                    ['url', $ep['handler_url'] ?? ''],
                ]);
                if ($i < count($domains) - 1) echo "\n";
            }
            echo "\n";
        } else {
            Output::info('No .formhandle config found. Run "formhandle init" to get started.');
        }
    }
}

File: rust/src/commands/status.rs
use serde_json::json;

use crate::{api, config, output};

pub fn run(is_json: bool) {
    let res = api::get("/");
    let cfg = config::read_config();

    if is_json {
        output::json_out(&json!({
            "api": res.data,
            "config": cfg,
        }));
        return;
    }

    output::heading("FormHandle API");

    if res.status == 200 {
        output::success("API is reachable");
        if let Some(s) = res.data.get("status").and_then(|v| v.as_str()) {
            output::info(&format!("Status: {}", s));
        }
        if let Some(v) = res.data.get("version").and_then(|v| v.as_str()) {
            output::info(&format!("Version: {}", v));
        }
    } else {
        output::error("API returned an unexpected status");
    }

    if let Some(c) = cfg {
        output::heading("Local Config (.formhandle)");
        let domains: Vec<&String> = c.keys().collect();
        for (i, domain) in domains.iter().enumerate() {
            println!("  {}", domain);
            let ep = &c[*domain];
            output::table(&[
                ("handler_id", ep.get("handler_id").and_then(|v| v.as_str()).unwrap_or("")),
                ("email", ep.get("email").and_then(|v| v.as_str()).unwrap_or("")),
                ("url", ep.get("handler_url").and_then(|v| v.as_str()).unwrap_or("")),
            ]);
            if i < domains.len() - 1 {
                println!();
            }
        }
        println!();
    } else {
        output::info("No .formhandle config found. Run \"formhandle init\" to get started.");
    }
}
